### PR TITLE
fix: reduce false positives in natural language activation/deactivation

### LINKED ISSUE #187: mode-tracker: natural language regex false positives on activation and deactivation
## Problem

The natural language activation/deactivation regexes in `hooks/caveman-mode-tracker.js` (added in #120) match too broadly, causing caveman to activate or deactivate on unrelated prompts.

### False activation

Any prompt containing "start" and "caveman" anywhere triggers activation, even when "caveman" is not the object of the action:

- `"How do I start a caveman fire?"` → activates caveman
- `"I want to start building a caveman-themed game"` → activates caveman
- `"Can you start explaining caveman lifestyle?"` → activates caveman

The root cause is `\b(activate|enable|turn on|start|talk like)\b.*\bcaveman\b` — the `.*` between "start" and "caveman" matches any intervening text.

### Negation ignored

`"don't activate caveman"` still activates. The negative guard checks for `stop|disable|turn off|deactivate` but not `don't`, `do not`, or `no`.

### False deactivation

`/\bnormal mode\b/i` matches without requiring "caveman" anywhere in the prompt:

- `"switch from normal mode to debug mode"` → deactivates caveman
- `"use normal mode for this task"` → deactivates caveman
- `"is this normal mode behavior?"` → deactivates caveman

Users discussing "normal mode" in any unrelated context lose caveman mode silently.

## Evidence

Tested the current regexes against 16 prompts — **9 failed (56% false positive rate)**. The failing cases are all prompts that a real user could plausibly type without intending to change caveman mode.

```
❌ "How do I start a caveman fire?"          → false activation
❌ "I want to start building a caveman game" → false activation
❌ "Can you start explaining caveman life?"  → false activation
❌ "don't activate caveman"                  → activation despite negation
❌ "switch from normal mode to debug mode"   → false deactivation
❌ "use normal mode for this task"           → false deactivation
❌ "is this normal mode behavior?"           → false deactivation
❌ "the normal mode of operation is..."      → false deactivation
```

## Suggested fix

1. Replace broad `start.*caveman` with exact `\bstart caveman\b` so "start a caveman fire" no longer matches
2. Require "caveman" alongside "normal mode" for deactivation: `(/\bnormal mode\b/i.test(prompt) && /\bcaveman\b/i.test(prompt))`
3. Add `don't`, `do not`, `no` to the negative guard

These three changes reduce the test failure rate from 9/16 to 0/19 (expanded test set) while preserving all intended activation phrases from the README.

## Platform
- [x] Claude Code
- [ ] Codex
- [ ] Other

### PR DESCRIPTION
**Pull request by shaun0927**
## The problem

The natural-language mode tracker in `hooks/caveman-mode-tracker.js` still matched too broadly around activation and deactivation phrases.

Three concrete failure modes prompted this PR:

1. **`start.*caveman` was too broad.** Prompts like `How do I start a caveman fire?` activated caveman even though `caveman` was not the action target.
2. **Negated activation still slipped through.** Phrases like `don't activate caveman` should not enable caveman.
3. **`normal mode` discussion caused false deactivation.** The tracker treated incidental mentions of `normal mode` as a stop command, so prompts like `switch from normal mode to debug mode` could silently disable caveman.

## What this PR changes

- Narrow activation matching so `start caveman` still works, but broader `start ... caveman` phrases do not.
- Keep negated activation blocked while allowing valid prompts such as `activate caveman, no markdown`.
- Preserve the documented stop phrase `normal mode`, but only as the exact standalone command instead of any incidental mention.
- Add regression tests for:
  - exact `start caveman` activation
  - no activation for `How do I start a caveman fire?`
  - no activation for `don't activate caveman`
  - valid activation for `activate caveman, no markdown`
  - no deactivation for incidental `normal mode` discussion
  - deactivation for exact `normal mode`

## Verification

- `python -m unittest discover -s tests -p 'test_*.py'`
- Targeted local hook repros for:
  - `How do I start a caveman fire?`
  - `start caveman`
  - `don't activate caveman`
  - `activate caveman, no markdown`
  - `switch from normal mode to debug mode`
  - `normal mode`

## Note

`tests/verify_repo.py` still fails on an existing activation-banner string expectation unrelated to this PR.

Closes #187

### CODE REVIEW

File: hooks/caveman-mode-tracker.js [hunk 1]
```diff
@@ -60,10 +66,12 @@ process.stdin.on('end', () => {
...
     }
 
-    // Detect deactivation — natural language and slash commands
+    // Detect deactivation — natural language and slash commands.
+    // "normal mode" requires "caveman" in the same prompt to avoid false
+    // deactivation when users discuss unrelated "normal mode" topics.
     if (/\b(stop|disable|deactivate|turn off)\b.*\bcaveman\b/i.test(prompt) ||
         /\bcaveman\b.*\b(stop|disable|deactivate|turn off)\b/i.test(prompt) ||
-        /\bnormal mode\b/i.test(prompt)) {
+        (/\bnormal mode\b/i.test(prompt) && /\bcaveman\b/i.test(prompt))) {
```
[the remark this answers is not in the document]
Comment (shaun0927) [reply hooks/caveman-mode-tracker.js]: Addressed in follow-up commit `58fa2f6`.

I kept `normal mode` as the documented standalone stop command, but narrowed it to the exact command so incidental mentions no longer deactivate caveman. Added regression tests covering:
- `normal mode` -> deactivates
- `switch from normal mode to debug mode` -> no deactivation
- `use normal mode for this task` -> no deactivation
- `is this normal mode behavior?` -> no deactivation

File: hooks/caveman-mode-tracker.js [hunk 2]
```diff
@@ -20,9 +20,15 @@ process.stdin.on('end', () => {
...
-      if (!/\b(stop|disable|turn off|deactivate)\b/i.test(prompt)) {
+    //
+    // Guard: require "caveman" to appear as a direct object of the action verb,
+    // not as a modifier in an unrelated clause ("start a caveman fire" should NOT
+    // activate). Also reject prompts containing negation words before the trigger.
+    if (/\b(activate|enable|turn on)\b.*\bcaveman\b/i.test(prompt) ||
+        /\btalk like\b.*\bcaveman\b/i.test(prompt) ||
+        /\bstart caveman\b/i.test(prompt) ||
+        /\bcaveman\b.*\b(mode|activate|enable|turn on)\b/i.test(prompt)) {
+      if (!/\b(stop|disable|turn off|deactivate|don't|do not|no)\b/i.test(prompt)) {
```
[the remark this answers is not in the document]
Comment (shaun0927) [reply hooks/caveman-mode-tracker.js]: Addressed by the current head.

The earlier follow-up removed bare `no` from the negation blocklist, and commit `58fa2f6` adds a regression test for `activate caveman, no markdown` so that case stays valid.

**Comment by shaun0927** [comment]
Follow-up pushed in `58fa2f6`.

This tightens `normal mode` deactivation to the exact standalone command, keeps the reviewed `no markdown` activation case valid, and adds regression tests for the natural-language edge cases discussed in review.

Verification:
- `python -m unittest discover -s tests -p 'test_*.py'`
- targeted local hook repros for start/no/don't/normal-mode prompts

Note: `python tests/verify_repo.py` still has a pre-existing activation-banner expectation mismatch unrelated to this PR.

**Comment by JuliusBrussee** [comment]
Thanks @shaun0927 — the shared parser now handles these false positives and punctuation cases. Closing as implemented.
